components/decorateMapComponent.js

```diff
@@ -48,8 +48,8 @@ export default function decorateMapComponent(Component, { componentType, provide
     if (components[provider]) return components[provider];
 
     if (provider === PROVIDER_DEFAULT) {
-      components.default = getDefaultComponent();
-      return components.default;
+      components[PROVIDER_DEFAULT] = getDefaultComponent();
```

Comment:
Don't need to check if already set, because it was then it would have been returned earlier.